### PR TITLE
login logic added

### PR DESCRIPTION
**Pull request by DmGrt**

### CODE REVIEW

**Review by RomanHreits — APPROVED** [review]
Well done!

File: src/main/java/internet/shop/controller/user/LoginController.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String login = req.getParameter("login");
+        String pwd = req.getParameter("pwd");
+        try {
+            User user = authenticationService.login(login, pwd);
+            HttpSession session = req.getSession();
+            session.setAttribute("user_id", user.getId());
```
Comment (dmytro-hryhoruk) [inline src/main/java/internet/shop/controller/user/LoginController.java]: let's make user_id a constant as in filter

File: src/main/java/internet/shop/dao/impl/UserDaoImpl.java [hunk 2]
```diff
@@ -40,4 +40,11 @@ public User update(User user) {
     public boolean delete(Long id) {
         return Storage.users.removeIf(user -> user.getId().equals(id));
     }
+
+    @Override
+    public Optional<User> findByLogin(String login) {
+        return Storage.users.stream()
+                .filter(s -> s.getLogin().equals(login))
```
Comment (dmytro-hryhoruk) [inline src/main/java/internet/shop/dao/impl/UserDaoImpl.java]:
```suggestion
                .filter(u -> u.getLogin().equals(login))
```

File: src/main/java/internet/shop/security/impl/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import internet.shop.exceptions.AuthenticationException;
+import internet.shop.lib.Inject;
+import internet.shop.lib.Service;
+import internet.shop.model.User;
+import internet.shop.security.AuthenticationService;
+import internet.shop.service.UserService;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject private UserService userService;
```
Comment (dmytro-hryhoruk) [inline src/main/java/internet/shop/security/impl/AuthenticationServiceImpl.java]:
```suggestion
    @Inject 
    private UserService userService;
```
for me in this way it's more readable especially that in other services you put annotation above fields

File: src/main/java/internet/shop/security/impl/AuthenticationServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        User userFromDB = userService.findByLogin(login).orElseThrow(() ->
+                new AuthenticationException("Incorrect username or password!"));
+
+        if (userFromDB.getPassword().equals(password)) {
+            return userFromDB;
+        }
+        throw new AuthenticationException("Incorrect username or password!");
```
Comment (dmytro-hryhoruk) [inline src/main/java/internet/shop/security/impl/AuthenticationServiceImpl.java]: either make both checks in single if or make exception message a constant

File: src/main/java/internet/shop/web/filters/AuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+            throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) request;
+        HttpServletResponse resp = (HttpServletResponse) response;
+        String url = req.getServletPath();
+        if (url.equals("/login") || url.equals("/user/registration")) {
+            chain.doFilter(req, resp);
+            return;
+        }
+        Long userId = (Long) req.getSession().getAttribute(USER_ID);
+        if (userService.get(userId) == null) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/internet/shop/web/filters/AuthenticationFilter.java]: let's first check if userId == null

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**